### PR TITLE
dyn_cfg: add dynamic config support for Grantor

### LINKED ISSUE #62: Add dynamic config support for Grantor
The dynamic configuration block only supports running on Gatekeeper servers, but we should enable it to run on Grantor servers as well.

This feature should remove the relevant `TODO` in `config/dynamic.c`.

### PR DESCRIPTION
**Pull request by mengxiang0811**
This patch closes #62

### CODE REVIEW

File: config/dynamic.c [hunk 1]
```diff
@@ -311,17 +342,52 @@ setup_dy_lua(lua_State *lua_state)
...
+static void
+send_new_lua_states(lua_State **lua_states,
+	int num_lua_states, struct gt_config *gt_conf)
+{
+	int i;
+
+	for (i = 0; i < num_lua_states; i++) {
+		int ret;
+		struct gt_cmd_entry *entry;
+		struct gt_instance *instance = &gt_conf->instances[i];
```
[the remark this answers is not in the document]
Comment (mengxiang0811) [reply config/dynamic.c]: I didn't see the scenario where we need ``num_lua_states`` greater than the #instances, could you please clarify this?

[the remark this answers is not in the document]
Comment (AltraMayor) [reply config/dynamic.c]: My point is just that `send_new_lua_states()` doesn't protect itself against a buggy caller. The very least is to have an `RTE_VERIFY()` to quickly catch those bugs.

File: include/gatekeeper_config.h [hunk 2]
```diff
@@ -106,14 +110,18 @@ struct dynamic_config {
 
 	/* Specify the receiving timeouts until reporting an error. */
 	struct timeval   rcv_time_out;
+
+	/* The lua states used to handle the dynamic configuration files. */
+	int              num_lua_states;
+	lua_State        **lua_states;
```
[the remark this answers is not in the document]
Comment (AltraMayor) [reply include/gatekeeper_config.h]: First of all, there should be only one Lua state to process a connection as the code in the repository already does. Second, there should be added a function available to the Lua script that comes through a connection to create these Lua states, load the GT configuration into these states, and send them to the GT blocks. The current patch is only loading the script that comes from the connection into these states and sending them to GT blocks.

[the remark this answers is not in the document]
Comment (AltraMayor) [reply include/gatekeeper_config.h]: By the way, what I mean by "GT configuration" in "load the GT configuration into these states" is the policy.

**Pull request closed without merging**